python/tvm/relax/transform/optimize_layout_transform.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, unused-argument, redefined-argument-from-local
+"""Relax Optimize Layout Transform pass."""
+from tvm.ir.module import IRModule
+from tvm.ir.transform import PassContext
+from tvm.relax import Expr, Function
+from tvm.relax.dpl import is_op, rewrite_call, wildcard
+from . import function_pass
+
+
+@function_pass(opt_level=0)
+class OptimizeLayoutTransform:
```

Comment:
The name "OptimizeLayoutTransform" sounds too generic for the simple thing it does. I'd suggest introducing "RemoveNop" pass and add this pattern as a first step.

Reply:
The reason for keeping the name as `OptimizeLayoutTransform` was to optimize two `layout_transform` operations into a single one when the index maps don't match. 
Right now the implementation is optimizing our redundant `layout_transform` operations when index_maps are matching. 
Eventually, we plan to extend this pass to converge two `transform_layout` operations into a single one.

Reply:
ok makes sense